code/script.js

```diff
@@ -222,96 +226,140 @@ const generateBoard = () => {
+    charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+  } else {
+    alert(`No, the character is not ${value} remove all ${value}s`)
+    charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+  }
   } else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
-    } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(`Yes, the person has ${value} ${category}! Keep all people with ${value} ${category}`)
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+    
+  } else {
+      alert(`No, the person doesnt have ${value} ${category}! Remove all people with ${value} ${category}`)
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
     }
   }
+    generateBoard();
+  }
 
   // Determine what is the category
   // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+  
+    // for hair and eyes :
+    //   charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
+    //   or
+    //   charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
 
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+    // for accessories and other
+    //   charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+    //   or
+    //   charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+
+// Invoke a function to redraw the board with the remaining people.
 
-  // Invoke a function to redraw the board with the remaining people.
-}
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
   // store the interaction from the player in a variable.
   // remember the confirm() ?
+  if (confirm(`Are you sure ${personToConfirm}`)) {
+    checkMyGuess(personToConfirm)
+  }
   // If the player wants to guess, invoke the checkMyGuess function.
 }
 
 // If you confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
+  board.style.display = "none"
+  if (personToCheck === secret.name) {
+    alert('Way to go')
+    winOrLoseText.innerHTML = 'Yeah'
+  } else {
+    alert('noooo')
+    winOrLoseText.innerHTML = 'Noooo'
+  }
+  winOrLose.style.display = 'flex'
+  board.style.display = 'none'
```

Comment:
You have the code here! I think you just had to change line 6 & 7 in your JS, the win/lose part would work!

You would also need to add one more global DOM selector: 
`const playAgain = getElementById('playAgain')`

As well as an eventlistener for the playAgain button:
`playAgain.addEventListener('click', start)`